### PR TITLE
(fix) O3-1676 & O3-1677: Workspace implementation improved as per design

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR brings the following changes to the workspaces:
### Workspace Variants
The registered workspaces can be registered with a variant keyword, which can be of 3 types:
#### Independent workspace (variant = "independent")
This variant denotes to the set of workspaces which are not collapsible and has a `X` button on the top right corner.
image

image

#### Siderail workspace (variant = "siderail")
These workspaces are allowed to callapse and marked as active on the corresponding action button on the side rail.
image

image

#### Clinical forms (variant = "clinical-form")
These workspaces are also allowed to collapse, but also can be maximised to full width and have a total width of 520px, as compared to the above 2 variants, which are 420px in width.
image
image

### Workspace width
The workspaces are now confined to 420px/ 520px, depending on the variant. The workspace width will remain fixed as the desktop size changes.

Reference: https://zeroheight.com/23a080e38/p/483a22-workspace

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-chart/assets/51502471/c08bc38c-5204-42af-ab6a-0a1487bcf01a

## Related Issue
https://issues.openmrs.org/browse/O3-1676
https://issues.openmrs.org/browse/O3-1677

### CODE REVIEW

**Review by hadijahkyampeire — APPROVED** [review]
LGTM, thanks @vasharma05

**Comment by brandones** [comment]
`variant` is a very general name for something that has very specific behavior. Semantically it seems redundant with "type." And you did not document it anywhere (the type declaration would have been the place). I would much rather you have added two booleans, `canMaximize` and `canHide`, and then you only show the close button if `canHide` is `false`.

**Comment by brandones** [comment]
@vasharma05 This cannot possibly be the correct window width.

image

**Comment by vasharma05** [comment]
> variant is a very general name for something that has very specific behavior. Semantically it seems redundant with "type." And you did not document it anywhere (the type declaration would have been the place). I would much rather you have added two booleans, canMaximize and canHide, and then you only show the close button if canHide is false.

Actually I tried to go via the available documentation itself. 
Documentation available here: https://zeroheight.com/23a080e38/p/483a22-workspace

**Comment by vasharma05** [comment]
If in the long run, the variant comes out to be an issue, I'll be happy to shift to your idea of 2 booleans.

**Comment by vasharma05** [comment]
For the order basket form, I'm going to ask Paul and Ciaran for the same.
Thanks for pointing it out.
Thanks!

File: packages/esm-patient-chart-app/src/workspace/workspace-window.scss [hunk 1]
```diff
@@ -61,26 +60,32 @@
...
+    .dynamicWidth {
+      width: 32.25rem;
     }
   }
-}
 
-/* Small desktop */
-:global(.omrs-breakpoint-lt-large-desktop){
-  .dynamicWidth {
-    width: calc(50% - 1.625rem);
```
Comment (pirupius) [inline packages/esm-patient-chart-app/src/workspace/workspace-window.scss:83]: Hey @vasharma05 are we able to revert the width size? the workspace has become really small to display form content correctly. 

>On desktop screens, the workspace sits on the right side of the page. Its width is 420px in most cases, and 520px if it's a clinical form. This width still allows actions to be performed with full visibility of information in the chart.

In the case that we're to follow the documentation, let's change the 26.25rem that represents the lower point of 420px, in favor of the upper limit of 520px (~32.5rem) to give more space

**Comment by brandones** [comment]
@vasharma05 Okay, let's look at what's happened. These are the designs:

image

This is what has been implemented:

image

It looks like you're right about the overall width, and the problem is that some fields are not sized correctly to display well. And the little "x" in the Dose Unit dropdown is probably superfluous and not helping either. Let's please fix the form fields so that the form is usable again.

**Comment by brandones** [comment]
@pirupius What form are you having trouble with? Could you send a screenshot or something?

File: packages/esm-patient-chart-app/src/workspace/workspace-window.scss (hunk 1 above)
Comment (vasharma05) [reply packages/esm-patient-chart-app/src/workspace/workspace-window.scss:83]: Variant: `clinical-form` will provide you with the width of 520px as I mentioned above as well.

You can provide the variant in either the `meta` of the extension registration, directly in the `registerWorkspace`. You can see the changed files for reference.
Thanks!

**Comment by vasharma05** [comment]
> It looks like you're right about the overall width, and the problem is that some fields are not sized correctly to display well. And the little "x" in the Dose Unit dropdown is probably superfluous and not helping either. Let's please fix the form fields so that the form is usable again.

Sure @brandones !
I'll create a ticket and fix this.
Thanks for the deep dive!

**Comment by pirupius** [comment]
@brandones only noticed it with the medication forms.

**Comment by vasharma05** [comment]
Hi @brandones @pirupius , please review the improvement to this PR: https://github.com/openmrs/openmrs-esm-patient-chart/pull/1333

**Pull request merged**